### ISSUE TITLE
Created server attempts to bootstrap against localhost, not server defined in knife.rb

### ISSUE THREAD

**Issue by mshappe** [chef/knife-ec2#266]
After some wrestling, I finally got knife ec2 to actually create an instance and start **trying** to boostrap it, only to hit a wall when it came to actually talking to a server.

In `$REPO/.chef/knife.rb`:
`chef_server_url          "https://ec2-54-183-138-149.us-west-1.compute.amazonaws.com/organizations/passare"
`

```
Connecting to ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Installing Chef Client...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com --2014-12-17 22:20:50--  https://www.opscode.com/chef/install.sh
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Resolving www.opscode.com (www.opscode.com)... 184.106.28.91
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Connecting to www.opscode.com (www.opscode.com)|184.106.28.91|:443... connected.
ec2-54-183-187-35.us-west-1.compute.amazonaws.com HTTP request sent, awaiting response... 200 OK
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Length: 18285 (18K) [application/x-sh]
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Saving to: ‘STDOUT’
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
100%[======================================>] 18,285      --.-K/s   in 0.001s
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com 2014-12-17 22:20:51 (23.2 MB/s) - written to stdout [18285/18285]
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Downloading Chef 12 for ubuntu...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com downloading https://www.opscode.com/chef/metadata?v=12&prerelease=false&nightlies=false&p=ubuntu&pv=14.04&m=x86_64
ec2-54-183-187-35.us-west-1.compute.amazonaws.com   to file /tmp/install.sh.1347/metadata.txt
ec2-54-183-187-35.us-west-1.compute.amazonaws.com trying wget...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com url    https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/13.04/x86_64/chef_12.0.3-1_amd64.deb
ec2-54-183-187-35.us-west-1.compute.amazonaws.com md5    76910abba3f7902d1b4d176f7a568c3a
ec2-54-183-187-35.us-west-1.compute.amazonaws.com sha256 3e172853aa0bcc27415a14b1013811dd44e5f523f4ba34714b66191ba365e796
ec2-54-183-187-35.us-west-1.compute.amazonaws.com downloaded metadata file looks valid...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com downloading https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/13.04/x86_64/chef_12.0.3-1_amd64.deb
ec2-54-183-187-35.us-west-1.compute.amazonaws.com   to file /tmp/install.sh.1347/chef_12.0.3-1_amd64.deb
ec2-54-183-187-35.us-west-1.compute.amazonaws.com trying wget...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Comparing checksum with sha256sum...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Installing Chef 12
ec2-54-183-187-35.us-west-1.compute.amazonaws.com installing with dpkg...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Selecting previously unselected package chef.
(Reading database ... 51091 files and directories currently installed.)
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Preparing to unpack .../chef_12.0.3-1_amd64.deb ...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Unpacking chef (12.0.3-1) ...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Setting up chef (12.0.3-1) ...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Thank you for installing Chef!
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Starting first Chef Client run...
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Starting Chef Client, version 12.0.3
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Creating a new client identity for dev-env-test42 using the validator key.
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:09+00:00] ERROR: Connection refused connecting to http://localhost:8889/clients, retry 1/5
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:14+00:00] ERROR: Connection refused connecting to http://localhost:8889/clients, retry 2/5
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:19+00:00] ERROR: Connection refused connecting to http://localhost:8889/clients, retry 3/5
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:24+00:00] ERROR: Connection refused connecting to http://localhost:8889/clients, retry 4/5
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:29+00:00] ERROR: Connection refused connecting to http://localhost:8889/clients, retry 5/5
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com ================================================================================
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Chef encountered an error attempting to create the client "dev-env-test42"
ec2-54-183-187-35.us-west-1.compute.amazonaws.com ================================================================================
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Network Error:
ec2-54-183-187-35.us-west-1.compute.amazonaws.com --------------
ec2-54-183-187-35.us-west-1.compute.amazonaws.com There was a network error connecting to the Chef Server:
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Connection refused - Connection refused connecting to http://localhost:8889/clients, giving up
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Relevant Config Settings:
ec2-54-183-187-35.us-west-1.compute.amazonaws.com -------------------------
ec2-54-183-187-35.us-west-1.compute.amazonaws.com chef_server_url  "http://localhost:8889"
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com If your chef_server_url is correct, your network could be down.
ec2-54-183-187-35.us-west-1.compute.amazonaws.com
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:34+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
ec2-54-183-187-35.us-west-1.compute.amazonaws.com Chef Client failed. 0 resources updated in 26.194483984 seconds
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:34+00:00] ERROR: Connection refused - Connection refused connecting to http://localhost:8889/clients, giving up
ec2-54-183-187-35.us-west-1.compute.amazonaws.com [2014-12-17T22:21:34+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by cskelly** [comment]
Hello, have you had any luck with this? I am still getting this error when I even add --config <PATH TO knife.rb>  to the knife bootstrap call

**Comment by mshappe** [comment]
@cskelly None. I have in fact given up on it. I'm writing my own scripts for automating instance creation and then bootstrapping them.

**Comment by cskelly** [comment]
Thanks @mshappe I appreciate the response. I am thinking of doing the same solution. If I end up finding a fix I will post on this thread :)

**Comment by NimishaS** [comment]
@mshappe, @cskelly, Sorry for the delayed response. We are looking at this issue.

**Comment by mshappe** [comment]
Thank you. While its been a fun exercise learning how to do it myself, I would just as soon be able to hand off something more turn-key to my eventual successors :-D

**Comment by kaustubh-d** [comment]
@mshappe 
I've tried ec2 server create with chef-server "https://ec2-54-183-138-149.us-west-1.compute.amazonaws.com/organizations/myorg" and it works fine(i.e server create + bootstrap process)

Your logs shows that  bootstrap is using **https://localhost:8889** as a **chef_server_url** which is nothing  but default **chef-zero** url.

Could you please check/share your knife.rb? 
Are you sure about bootstrap process is using right knife.rb config ? To check that you can try - `$ knife bootstrap <public_ip> -c ../knife.rb` or for windows - `$ knife bootstrap windows winrm  <public_ip> -c ../knife.rb`

**Comment by mshappe** [comment]
@kaustubh-d At the time I wrote this, I was sure of relatively little -- I'm still climbing the chef learning curve :-D

What was confusing me at the time was that it seemed to be OK if it bootstrapped Chef 11, but Chef 12 choked. That made me think that it was not a knife.rb problem.

I have since done other things that clean up my environment, so I'm willing to give it another whirl and se what happens.

**Comment by mshappe** [comment]
Aaand, yes, that was in fact the issue. I had several different `knife.rb`s floating around at one point as I experimented. I have since cleaned that up for unrelated reasons, but when @kaustubh-d made his suggestion above it occurred to me that it could indeed have been confusing the issue. Now I have exactly one `knife.rb` in `${HOME}/.chef` and voila, everything behaves.

Ah, well...I got to learn a lot about the Amazon API in the process of coding around it, at least :smile: 

Closing the issue. Thank you for your support.
